### DISCUSSION TITLE
Add search capability to output

### DISCUSSION THREAD

**Discussion by andrejohansson** [evidence-dev/evidence#2407] [General]
## Feature Description

I would like to have ability to search my generated page in the same way that the following projects do it:

- [Docusaurus Search Plugins](https://docusaurus.community/plugindirectory/?tags=search)
- [MdBook Search](https://github.com/ang-zeyu/infisearch)
  - [InfiniSearch Plugin](https://github.com/ang-zeyu/infisearch)

Please note that this may sound similar to #177 but in that ticket search was added to Evidence Documentation, not to the generated evidence page.

## Goal of Feature

Generating documentation with evidence data reports is nice, but documentation quickly grows big and requires search capabilities.

## Current Solution / Workarounds

No

## Alternatives

If evidence can be injected into Docusaurus instead of being the main owner of the layout.

**Comment by hughess** [comment]
I noticed Svelte has a nice search feature in their tutorial project:
https://learn.svelte.dev/tutorial/welcome-to-svelte

It uses their internal site-kit tools. We may be able to follow a similar approach:
https://github.com/sveltejs/site-kit/tree/master/packages/site-kit/src/lib/search

**Comment by archiewood** [comment]
We have added the ability to configure [Algolia Docsearch](https://docsearch.algolia.com/)

You can add this from a custom layout

https://docs.evidence.dev/themes-and-layouts/#custom-layout

**Reply by pax** [reply]
> We have added the ability to configure [Algolia Docsearch](https://docsearch.algolia.com/)

for available parameters, see #1646 or the [docs layout config](https://github.com/evidence-dev/evidence/blob/main/sites/docs/pages/%2Blayout.svelte)

**Comment by archiewood** [comment]
See: https://github.com/evidence-dev/evidence/pull/1646

**Comment by mcrascal** [comment]
Propose we keep this open for now -- algolia only works if your project is public and algolia accepts you into their docsearch program

**Comment by andrejohansson** [comment]
[Flexsearch](https://github.com/nextapps-de/flexsearch?tab=readme-ov-file) looks interesting. A sample integration with sveltekit can be [seen here](https://www.youtube.com/watch?v=llGupgNwmIk)

- https://joyofcode.xyz/blazing-fast-sveltekit-search

**Comment by mcrascal** [comment]
@andrejohansson how useful would search against just the markdown contents of your project be? Search that excludes the results of your queries is a much simpler proposition than search against the entire final results of the build that include the results of queries.

**Comment by andrejohansson** [comment]
Very very useful. 

You can see it yourself since you changed your docs to evidence. You include algolia there (which I assume doesn´t traverse the data either). 

Our page is an internal documentation site and therefore we cannot use Algolia or similar third party providers.

**Comment by andrejohansson** [comment]
@mcrascal how are your thoughts going regarding this? Anything I can do to help here?

**Comment by mcrascal** [comment]
@andrejohansson no updates - we want to add it, but haven't been able to prioritize it yet.  @ItsMeBrianD is quite interested in using https://github.com/askorama/orama when we do. 

Certainly we'd welcome a contribution if you're interested in working on it.

**Comment by ItsMeBrianD** [comment]
I did think of a way that may work for this:

We can likely iterate through the built pages in the evidence cli _after_ the vite build has finished and build the search indexes based on those results

**Comment by andrejohansson** [comment]
@ItsMeBrianD this plugin does something similar for docusaurus:
https://github.com/daldridge/docusaurus-plugin-lunr

**Comment by biscout42** [comment]
I have tried to integrate https://pagefind.app/ with evidence and it looks promising. The only issue I have at the moment - how-to adjust header and add another `<div/>`, similar to algolia. Any ideas are welcome.
